src/goto-programs/goto_program.cpp

```diff
@@ -117,9 +117,17 @@ std::ostream &goto_programt::output_instruction(
   case DECL:
   case DEAD:
   case FUNCTION_CALL:
-  case ASSIGN:
     out << from_expr(ns, identifier, instruction.get_code()) << '\n';
     break;
+  case ASSIGN:
+    out << from_expr(ns, identifier, instruction.get_code());
+    {
+      const irep_idt &commentw = instruction.source_location.get_comment();
+      if (!commentw.empty())
+        out << " // " << commentw;
+    }
+    out << '\n'; 
+    break;
```

Comment:
Why was this change needed?

Reply:
When we display a GOTO program by the command `goto-instrument --show-goto-functions`, I want to show a comment `// Evaluate and store the loop variant before the loop` next to the assignment for the first temporary variable, and likewise a comment `// Evaluate and store the loop variant after the loop` next to the second temporary variable's assignment. This is why I modified the function `output_instruction` in `goto_program.cpp` such that it can print out not only comments of assertions but also comments of assignments. If you think this feature is redundant, please let me know!

Reply:
We usually don't add comments on assignments, but only on the assertions (which you're doing already in code_contracts.cpp). The comments on assignments are not guaranteed to be preserved across subsequent transformations (simplification passes etc.), so may be we can remove these comments and revert this printing logic.

Reply:
OK. I will revert the change I had made.